### PR TITLE
Adds comment_count query to GET /api/articles/:article_id API

### PR DESCRIPTION
**Pull request by cdesale**
- Adds comment_count query to GET /api/articles/:article_id API
- Adds necessary tests
- Also updates endpoints.json file

### CODE REVIEW

File: endpoints.json [hunk 1]
```diff
@@ -97,5 +97,82 @@
...
+          "created_at": 1602828180000,
+          "article_img_url": "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+        }
+      ]
+    }
+  },
```
Comment (nc-mick-fay) [inline endpoints.json]: This is an added query to a pre-existing endpoint, you can just add it to that endpoint's description

**Review by nc-mick-fay — APPROVED** [review]
I like that you've given an overall description of your pull request when you made it - it makes it really clear what I'm looking at!

Great changes to your query and use of `::int` to cant your count into a number.

When following Test Driven Development, it's bad practice to go back and change old tests. Ideally, you would write a new test for the new functionality of adding a comment count. Your tests act as documentation, so by adding a new test for this we would see the development of a new feature, rather than one line of code that can be easily missed.

Your topic query can also be added to the queries array in your /api/articles/:article_id endpoint description in your JSON file, you don't need to make a totally new one

Feel free to merge when you're ready after these small changes are made

File: __tests__/get-article-by-id.test.js [hunk 2]
```diff
@@ -33,6 +32,25 @@ describe("GET /api/articles/:article_id", () => {
...
       });
   });
+    test("200: responds with the article object containing additional property called comment_count for the given article id", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          const { article } = body;
+          expect(article.article_id).toBe(1);
+          expect(article.comment_count).toBe(11);
```
Comment (popatre) [inline __tests__/get-article-by-id.test.js:42]: Much better - could probably have just checked for comment count, as you already have a test checking for the other stuff.

**Review by popatre — APPROVED** [review]
Looks good - please see comment about testing though - just for future reference

**Pull request merged**